### DISCUSSION TITLE
Add a rust-bitcoin book

### DISCUSSION THREAD

**Discussion by sanket1729** [rust-bitcoin/rust-bitcoin#1241] [General]
Is it worth adding a tutorial/book like the rust-book? 

I was thinking of doing this post 1.0, but maybe having a book before it might also be valuable. 

This can be used as
- Getting started
- Tutorials for common applications
- Onboarding new students via the Summer of the bitcoin program.

**Comment by tcharding** [comment]
I wouldn't want to spend any time on this myself pre-v1.0

**Reply by apoelstra** [reply]
I agree with both "not yet" and "I don't want to do the work" :P 

But more seriously, yeah, I think this is a great idea.

**Reply by Kixunil** [reply]
How about during RC phase of 1.0? I think it'd be the best time to polish it.

**Comment by DanGould** [comment]
I think one of the easiest ways to create something of value would be to translate @nicolasdorier's [programming the blockchain in c#](https://programmingblockchain.gitbook.io/programmingblockchain/) to rust. Maybe rename to programming bitcoin in rust

**Reply by apoelstra** [reply]
I spot checked the first couple sections and I think it probably doesn't align well with our goals -- the introduction defending "blockchain" over "bitcoin" and "C#" over "anything else" okay, we can change :) but even once we get into the technicals I see that the "Address" section
* is very specific to single-key legacy addresses with no mention of other common address types
* similarly talks about base58 but not base32
* talks about EC keys in a way that's low-level enough to be overwhelming but not low-level enough to be useful
* says that there are "two bitcoin networks', ignoring regtest etc

I think these mostly all stem from the book being very old, and maybe written by a single author with a narrow focus, but these are still good reasons not to use it as a basis.
